effects/mk-git-branch.nix

```diff
@@ -0,0 +1,58 @@
+/* Push a branchRoot to a git remote. Useful for deploying content via
+   git.
+
+   Example:
+     mkGitBranchViaEffect {
+       pushToBranch = "staging-myWebsite";
+       preGitInit = ''
+         echo "staging-myWebsite.example.com" > .domains
+       '';
+       branchRoot = "${./myWebsiteRoot}";
+       gitRemote = "git@codeberg.org";
+       hostKey = "codeberg.org ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIIVIC02vnjFyL+I4RHfvIGNtOgJMe769VTF1VR4EB3ZB";
+       owner = "myName";
+       repo = "pages";
+       committerEmail = "edolstra@gmail.com";
+       committerName = "Hercules-CI Effects";
+       authorName = "Hercules-CI Effects";
+     };
+*/
+{
+  mkEffect,
+  git,
+  openssh,
+}:
+args@{
+  gitRemote,
+  hostKey,
+  pushToBranch,
+  owner,
+  repo,
+  branchRoot,
+  committerEmail,
+  committerName,
+  authorName,
+  preGitInit ? "",
+  sshSecretName ? "ssh",
+  ...
+}:
+mkEffect {
+  inputs = [ openssh git ];
+  secretsMap = {
+    "ssh" = ${sshSecretName};
+  };
```

Comment:
I guess the value of this is to force setting an `ssh` secret, but what if the user wants to use `https` instead?